src/pk-main.c

```diff
@@ -183,6 +183,11 @@ main (int argc, char *argv[])
 
 	/* after how long do we timeout? */
 	exit_idle_time = g_key_file_get_integer (conf, "Daemon", "ShutdownTimeout", NULL);
+        /* THIS COMMENT IS A TSUNAMI STONE 
+         * Before removing the default timeout, please study the git history and
+         * be sure that you are not regressing Redhat bugzilla #1354074 (again). */
+	if (exit_idle_time == 0)
+		exit_idle_time = 300;
```

Comment:
Let's drop that comment:

```suggestion
	if (exit_idle_time == 0)
		exit_idle_time = 300;
```

Reply:
Let's not.  History speaks to its necessity, and I don't want this coming back again a few years down the road.

Reply:
After dropping the change it's an inaccurate comment though, and even worse, it's a Fedora/DNF-specific workaround to an entirely different issue. So there's not much of a point to keeping a random comment in this place.
A comment with "there was a bug once for which there was a workaround here once" is quite odd...

Reply:
So here you are adding a comment that says "Before removing the default timeout" in the same commit that itself removes the default timeout. Hence, the new comment is immediately stale and incorrect. You have to at least edit it somehow to not be wrong.

Reply:
Could you clarify "after dropping the change"? Which change, and how is the comment inaccurate without it?  @gordonmessmer's DNF plugin solves the problem of packagekit's model of the system getting out of sync with dnf, but if someone in the future comes up with some *other* reason they think packagekitd should run forever, they could easily reintroduce the resource squandering.

The magnitude of the issue is DNF-specific, but I don't think the issue itself is.  The cost of packagekitd is particularly high with the DNF backend, but with other backends it is not zero, as you said last week.

Reply:
@mcatanzaro I am adding a comment that says "Before removing the default timeout" in the same commit that **adds back** the default timeout.  300s is the default timeout. Removing the default timeout is something that future maintainers should not do without great caution.

However if you misunderstood it, it is not clear, so I will word it.

Reply:
> Could you clarify "after dropping the change"? Which change, and how is the comment inaccurate without it?

With your change, the daemon will quit again. So the behavior the text speaks about is "times out now", which is not what it originally was about. With the status quo changed back again, that comment would be incredibly misleading.

> The magnitude of the issue is DNF-specific, but I don't think the issue itself is. The cost of packagekitd is particularly high with the DNF backend, but with other backends it is not zero, as you said last week.

It's pretty DNF-specific, the other backends consume memory, but nothing nearly this bad. But precisely because we want to consume as few background resources as possible, we *want* PK to time out again.

> Removing the default timeout is something that future maintainers should not do without great caution.

Honestly, I think you should either drop the previous comment, or reword it to say that the timeout should not be dropped because backends may leak resources.